runtimes/protocol/lsp.ts

```diff
@@ -86,12 +86,16 @@ export interface AWSInitializationOptions {
      */
     clientInfo?: ExtendedClientInfo
     /**
-     * Client data directory, used as an application data directory to store LSP server-specific data.
-     * LSP servers are expected to use the directory for storing their configuration files,
-     * temporary data, and user-specific settings.
-     * The directory is extected to maintain data across updates and re-installations.
+     * Client data folder, used as an application data folder to store LSP server-specific data.
+     * LSP servers are expected to use the folder for storing their configuration files,
+     * temporary data, and user-specific settings. Servers should create subfolder
+     * within the client data folder for storing server-specific data.
+     *
+     * The folder is expected to be created by client, separate folder per application
+     * (e.g. multiple IDEs on a client machine should not share the same folder),
```

Comment:
I don't know that it needs to be addressed in the comments here, but we should consider that there is going to be different kinds of data in this folder, some long-lived and shared across all running instances of a particular IDE/destination to short-lived (IDE instance session or Flare process lifetime) and specific to IDE instance/session data.  

Presumably the implementing server will know which of those types of data it is working with.  In the case of the short-lived/specific to instance data, do we have a sufficient way for servers to tie the temporal data to an IDE instance (this is the case even if the Flare server is not shared between IDE instances)?  I see InitializeParams has an optional processId property that can be passed by the client.  That may be sufficient if the client provides it.

Reply:
Temp folder can be better place for short-lived data.